controllers/argocd/applicationset/role.go

```diff
@@ -80,6 +80,9 @@ func (asr *ApplicationSetReconciler) reconcileRole() error {
 
 func (asr *ApplicationSetReconciler) deleteRole(name, namespace string) error {
 	if err := permissions.DeleteRole(name, namespace, asr.Client); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
```

Comment:
is this change necessary here? seems unrelated to server package
also not found errors are already handled within `permissions.DeleteRole`

Reply:
I removed not found error handling in `permissions.DeleteRole` and in some other functions because it was getting difficult to handle logging in controller code where the delete function is called repeatedly on each reconciliation.  For example in case of server role delete, the first delete call would log role deleted message correctly but subsequent calls to delete were logging the same delete message even though it was just exiting without performing a actual delete of the object. 
I couldn't think of a better way to handle it. Please let me know if there's a different approach we could consider. Also if we are okay with having false positive in logs, we can revert this change.

Reply:
An alternative approach could be to avoid relying on `permissions.DeleteRole` and similar delete functions where repeated calls are made. Instead, we could handle the delete explicitly using `client.Delete`.

Reply:
>the first delete call would log role deleted message correctly but subsequent calls to delete were logging the same delete message even though it was just exiting without performing a actual delete of the object.

makes sense